Source/WebCore/crypto/mac/CryptoAlgorithmAES_GCMMac.cpp

```diff
@@ -43,12 +44,20 @@ ALLOW_DEPRECATED_DECLARATIONS_BEGIN
 ALLOW_DEPRECATED_DECLARATIONS_END
     if (status)
         return Exception { ExceptionCode::OperationError };
-
     memcpy(cipherText.data() + plainText.size(), tag.data(), desiredTagLengthInBytes);
 
     return WTFMove(cipherText);
 }
 
+static ExceptionOr<Vector<uint8_t>> encryptCryptoKitAES_GCM(const Vector<uint8_t>& iv, const Vector<uint8_t>& key, const Vector<uint8_t>& plainText, const Vector<uint8_t>& additionalData, size_t desiredTagLengthInBytes)
+{
+    Vector<uint8_t> cipherText(plainText.size() + 16);
+    if (CkAesGcm::encrypt(key.data(), key.size(), iv.data(), iv.size(), additionalData.data(), additionalData.size(), plainText.data(), plainText.size(), cipherText.data()))
```

Comment:
Since this file is already Cocoa-only, can we turn it into an ObjC++ file (.mm) and call the Swift shim directly? That ought to eliminate the C++ shim in `CryptoKitAesShim.{h,mm}`, so we'll only need one shim instead of two :)

Reply:
great idea. Relentless code reduction appetite, I like it!

Reply:
done. the `mm/h` shim is gone now.